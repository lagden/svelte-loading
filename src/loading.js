/* istanbul ignore file */

import * as acts from './acts.js'
import Loading from './Loading.svelte'

export {
	Loading,
	acts,
}
