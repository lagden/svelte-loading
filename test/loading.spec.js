import {
	describe,
	expect,
	test,
	afterEach,
} from '@jest/globals'
import {cleanup, render} from '@testing-library/svelte'
import {Loading, acts} from '../src/loading.js'

describe('Loading default', () => {
	afterEach(cleanup)

	test('should match snapshot', () => {
		acts.show(true)

		const {container} = render(Loading, {
			props: {
				class: 'teste',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Loading Jelly', () => {
	afterEach(cleanup)

	test('should match snapshot', () => {
		acts.show(true)

		const {container} = render(Loading, {
			props: {
				class: 'teste',
				animation: 'Jelly',
			},
		})
		expect(container).toMatchSnapshot()
	})
})
