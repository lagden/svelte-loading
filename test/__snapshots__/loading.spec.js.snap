// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Jelly should match snapshot 1`] = `
<body>
  <div>
    <div
      class="_tadashi_svelte_loading teste svelte-1ej2r2y"
    >
      <div
        class="jelly svelte-1qkf8ue"
        style="--uib-color: black;"
      />
       
      <svg
        class="jelly-maker svelte-1qkf8ue"
        height="0"
        width="0"
      >
        <defs>
          <filter
            id="uib-jelly-ooze"
          >
            <fegaussianblur
              in="SourceGraphic"
              result="blur"
              stdDeviation="6.25"
            />
            <fecolormatrix
              in="blur"
              mode="matrix"
              result="ooze"
              values="1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 18 -7"
            />
            <feblend
              in="SourceGraphic"
              in2="ooze"
            />
          </filter>
        </defs>
      </svg>
    </div>
    
  </div>
</body>
`;

exports[`Loading default should match snapshot 1`] = `
<body>
  <div>
    <div
      class="_tadashi_svelte_loading teste svelte-1ej2r2y"
    >
      <svg
        class="ring svelte-u9avcq"
        stroke-width="5"
        style="--uib-color:black;"
        viewBox="25 25 50 50"
      >
        <circle
          class="svelte-u9avcq"
          cx="50"
          cy="50"
          r="20"
        />
      </svg>
    </div>
    
  </div>
</body>
`;
